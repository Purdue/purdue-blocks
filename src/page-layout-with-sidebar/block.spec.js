// title-nav test
jest.setTimeout(30000)
import {
    createNewPost,
    enablePageDialogAccept,
    getEditedPostContent,
    insertBlock,
} from '@wordpress/e2e-test-utils';

import {
    blockStartup,
    clickCheckbox,
    clickRadio,
} from '../test-helpers'

const block = {blockTitle: 'Page Layout With Sidebar', blockName: 'purdue-blocks/page-layout-with-sidebar'}

describe( 'ðŸ”¬ Page Layout With Sidebar Block', () => {
    beforeAll( async () => {
        await enablePageDialogAccept();
    } );
    beforeEach( async () => {
        await createNewPost();
    } );

    test( 'Block should be available.', async () => {
        await insertBlock( block.blockTitle )


        // tests that the block is properly inserted and matches the existing snapshot
        expect(await page.$(`[data-type="${block.blockName}"]`)).not.toBeNull()
        expect( await getEditedPostContent() ).toMatchSnapshot()
    } )

    describe('ðŸ”¬ Side Panel Settings', () => {
        test('ðŸ”Ž Sidebar Location On Desktop Options', async () => {
            await blockStartup(block)

            await clickRadio('Sidebar Location On Desktop', 'left')

            let editedContent = await getEditedPostContent()
            expect( editedContent.includes(`"sidebarLocationDesktop":"right"`)).toBe(false)
            expect( await getEditedPostContent() ).toMatchSnapshot()


            await clickRadio('Sidebar Location On Desktop', 'right')

            editedContent = await getEditedPostContent()
            expect( editedContent.includes(`"sidebarLocationDesktop":"right"`)).toBe(true)
            expect( await getEditedPostContent() ).toMatchSnapshot()
        })
        test('ðŸ”Ž Sidebar Location On Mobile Options', async () => {
            await blockStartup(block)

            await clickRadio('Sidebar Location On Mobile', 'above')

            let editedContent = await getEditedPostContent()
            expect( editedContent.includes(`"sidebarLocationMobile":"below"`)).toBe(false)
            expect( await getEditedPostContent() ).toMatchSnapshot()


            await clickRadio('Sidebar Location On Mobile', 'right')

            editedContent = await getEditedPostContent()
            expect( editedContent.includes(`"sidebarLocationMobile":"below"`)).toBe(true)
            expect( await getEditedPostContent() ).toMatchSnapshot()
        })
    })
})