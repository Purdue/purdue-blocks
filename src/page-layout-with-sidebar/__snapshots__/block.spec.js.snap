// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`🔬 Page Layout With Sidebar Block Block should be available. 1`] = `
"<!-- wp:purdue-blocks/page-layout-with-sidebar -->
<div class=\\"section\\"><div class=\\"container\\"><!-- wp:columns {\\"className\\":\\"page-layout columns is-multiline\\"} -->
<div class=\\"wp-block-columns page-layout columns is-multiline\\"><!-- wp:column {\\"className\\":\\"column is-8\\"} -->
<div class=\\"wp-block-column column is-8\\"><!-- wp:paragraph {\\"placeholder\\":\\"Main content area. Start typing to add content, or remove this default paragraph block and then add new blocks.\\"} -->
<p></p>
<!-- /wp:paragraph --></div>
<!-- /wp:column -->

<!-- wp:column {\\"className\\":\\"column is-3\\"} -->
<div class=\\"wp-block-column column is-3\\"><!-- wp:paragraph {\\"placeholder\\":\\"Sidebar content area. Start typing to add content, or remove this default paragraph block and then add new blocks.\\"} -->
<p></p>
<!-- /wp:paragraph --></div>
<!-- /wp:column --></div>
<!-- /wp:columns --></div></div>
<!-- /wp:purdue-blocks/page-layout-with-sidebar -->"
`;

exports[`🔬 Page Layout With Sidebar Block 🔬 Side Panel Settings 🔎 Sidebar Location On Desktop Options 1`] = `
"<!-- wp:purdue-blocks/page-layout-with-sidebar {\\"sidebarLocationDesktop\\":\\"left\\"} -->
<div class=\\"section desktop-reverse\\"><div class=\\"container\\"><!-- wp:columns {\\"className\\":\\"page-layout columns is-multiline\\"} -->
<div class=\\"wp-block-columns page-layout columns is-multiline\\"><!-- wp:column {\\"className\\":\\"column is-8\\"} -->
<div class=\\"wp-block-column column is-8\\"><!-- wp:paragraph {\\"placeholder\\":\\"Main content area. Start typing to add content, or remove this default paragraph block and then add new blocks.\\"} -->
<p></p>
<!-- /wp:paragraph --></div>
<!-- /wp:column -->

<!-- wp:column {\\"className\\":\\"column is-3\\"} -->
<div class=\\"wp-block-column column is-3\\"><!-- wp:paragraph {\\"placeholder\\":\\"Sidebar content area. Start typing to add content, or remove this default paragraph block and then add new blocks.\\"} -->
<p></p>
<!-- /wp:paragraph --></div>
<!-- /wp:column --></div>
<!-- /wp:columns --></div></div>
<!-- /wp:purdue-blocks/page-layout-with-sidebar -->"
`;

exports[`🔬 Page Layout With Sidebar Block 🔬 Side Panel Settings 🔎 Sidebar Location On Desktop Options 2`] = `
"<!-- wp:purdue-blocks/page-layout-with-sidebar -->
<div class=\\"section\\"><div class=\\"container\\"><!-- wp:columns {\\"className\\":\\"page-layout columns is-multiline\\"} -->
<div class=\\"wp-block-columns page-layout columns is-multiline\\"><!-- wp:column {\\"className\\":\\"column is-8\\"} -->
<div class=\\"wp-block-column column is-8\\"><!-- wp:paragraph {\\"placeholder\\":\\"Main content area. Start typing to add content, or remove this default paragraph block and then add new blocks.\\"} -->
<p></p>
<!-- /wp:paragraph --></div>
<!-- /wp:column -->

<!-- wp:column {\\"className\\":\\"column is-3\\"} -->
<div class=\\"wp-block-column column is-3\\"><!-- wp:paragraph {\\"placeholder\\":\\"Sidebar content area. Start typing to add content, or remove this default paragraph block and then add new blocks.\\"} -->
<p></p>
<!-- /wp:paragraph --></div>
<!-- /wp:column --></div>
<!-- /wp:columns --></div></div>
<!-- /wp:purdue-blocks/page-layout-with-sidebar -->"
`;

exports[`🔬 Page Layout With Sidebar Block 🔬 Side Panel Settings 🔎 Sidebar Location On Mobile Options 1`] = `
"<!-- wp:purdue-blocks/page-layout-with-sidebar {\\"sidebarLocationMobile\\":\\"above\\"} -->
<div class=\\"section mobile-reverse\\"><div class=\\"container\\"><!-- wp:columns {\\"className\\":\\"page-layout columns is-multiline\\"} -->
<div class=\\"wp-block-columns page-layout columns is-multiline\\"><!-- wp:column {\\"className\\":\\"column is-8\\"} -->
<div class=\\"wp-block-column column is-8\\"><!-- wp:paragraph {\\"placeholder\\":\\"Main content area. Start typing to add content, or remove this default paragraph block and then add new blocks.\\"} -->
<p></p>
<!-- /wp:paragraph --></div>
<!-- /wp:column -->

<!-- wp:column {\\"className\\":\\"column is-3\\"} -->
<div class=\\"wp-block-column column is-3\\"><!-- wp:paragraph {\\"placeholder\\":\\"Sidebar content area. Start typing to add content, or remove this default paragraph block and then add new blocks.\\"} -->
<p></p>
<!-- /wp:paragraph --></div>
<!-- /wp:column --></div>
<!-- /wp:columns --></div></div>
<!-- /wp:purdue-blocks/page-layout-with-sidebar -->"
`;

exports[`🔬 Page Layout With Sidebar Block 🔬 Side Panel Settings 🔎 Sidebar Location On Mobile Options 2`] = `
"<!-- wp:purdue-blocks/page-layout-with-sidebar -->
<div class=\\"section\\"><div class=\\"container\\"><!-- wp:columns {\\"className\\":\\"page-layout columns is-multiline\\"} -->
<div class=\\"wp-block-columns page-layout columns is-multiline\\"><!-- wp:column {\\"className\\":\\"column is-8\\"} -->
<div class=\\"wp-block-column column is-8\\"><!-- wp:paragraph {\\"placeholder\\":\\"Main content area. Start typing to add content, or remove this default paragraph block and then add new blocks.\\"} -->
<p></p>
<!-- /wp:paragraph --></div>
<!-- /wp:column -->

<!-- wp:column {\\"className\\":\\"column is-3\\"} -->
<div class=\\"wp-block-column column is-3\\"><!-- wp:paragraph {\\"placeholder\\":\\"Sidebar content area. Start typing to add content, or remove this default paragraph block and then add new blocks.\\"} -->
<p></p>
<!-- /wp:paragraph --></div>
<!-- /wp:column --></div>
<!-- /wp:columns --></div></div>
<!-- /wp:purdue-blocks/page-layout-with-sidebar -->"
`;
